File: Challenge2-Finding Pi/piFinder.cc
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;

double randZeroToOne()
{
    return rand() / (RAND_MAX + 1.0);
}

double piFinder(int n)
{
  int circpts=0;
  for(int i=0 ;i<n;i++)
  {
    double x=randZeroToOne();
    double y=randZeroToOne();
    double ans=(x*x)+(y*y);
    if(ans<=1)
    {
      circpts++;
    }

  }
  return 4*(circpts/static_cast<double>(n));
}

int main()
{
  int n;
  cout<<"Enter number of points"<<endl;
  cin>>n;
  cout<<(piFinder(n))<<endl;
  return 0;
}

File: Finding Pi/piFinder.py
import random;


def piFinder():
    n=int(input("Enter the number of random points\n"))
    circpts=0;


    for i in range(n):
        x=random.random();
        y=random.random()
        h= (x**2) + (y**2)
        if h<=1:
            circpts+=1
    
    return 4 * (circpts/n)

print(piFinder())
